machineconfiguration/v1/types_machineosbuild.go

```diff
@@ -0,0 +1,217 @@
+	// building indicates that the build has been kicked off with the specified image builder
+	MachineOSBuilding BuildProgress = "Building"
+	// failed indicates that during the build or preparation process, the build failed.
+	MachineOSBuildFailed BuildProgress = "Failed"
+	// interrupted indicates that the user stopped the build process by modifying part of the build config
+	MachineOSBuildInterrupted BuildProgress = "Interrupted"
+	// succeeded indicates that the build has completed and the image is ready to roll out.
+	MachineOSBuildSucceeded BuildProgress = "Succeeded"
+)
+
+// Refers to the name of a rendered MachineConfig (e.g., "rendered-worker-ec40d2965ff81bce7cd7a7e82a680739", etc.):
+// the build targets this MachineConfig, this is often used to tell us whether we need an update.
+type MachineConfigReference struct {
+	// name is the name of the rendered MachineConfig object.
+	// This value should be between 10 and 253 characters, and must contain only lowercase
+	// alphanumeric characters, hyphens and periods, and should start and end with an alphanumeric character.
+	// +kubebuilder:validation:MinLength:=10
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +required
+	Name string `json:"name"`
+}
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
+	// group of the referent.
+	// The name must contain only lowercase alphanumeric characters, '-' or '.' and start/end with an alphanumeric character.
+	// Example: "", "apps", "build.openshift.io", etc.
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +required
+	Group string `json:"group"`
+	// resource of the referent.
+	// This value should consist of at most 63 characters, and of only lowercase alphanumeric characters and hyphens,
+	// and should start and end with an alphanumeric character.
+	// Example: "deployments", "deploymentconfigs", "pods", etc.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	Resource string `json:"resource"`
+	// namespace of the referent.
+	// This value should consist of at most 63 characters, and of only lowercase alphanumeric characters and hyphens,
+	// and should start and end with an alphanumeric character.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
```

Comment:
This is a label, not a domain, so the regex here is not correct
```suggestion
	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])$`
```

Reply:
Fixed